### ISSUE TITLE
Reproducing the results of the paper

### ISSUE THREAD

**Issue by vigna** [caba5/WebGraph#6]
Me and @colobrodo we're trying to reproduce the results of the paper. We are trying to use `compress_zuckerli` on a few graphs used in the paper, but we are using the default webgraph parameters and we get consistently larger graphs. We think there's a specific selection of parameters of the CLI that should be used, but these choices are scattered through the paper and hard to follow.

Do you have a reproducibility script, or at least could you suggest us the CLI lines to get graphs compressed as in the paper? Thanks!

**Comment by caba5** [comment]
Hi @vigna and @colobrodo, sorry for the late reply.
Unfortunately, there is no reproducibility script, as the main aim of the project was not to prove nor disprove the results published in the authors' papers, but instead to do some experiments on them and mainly to rewrite them in Rust, testing the language capabilities.
Zuckerli's paper is cryptic in the results-reporting section, as it does specify some used parameters, but fails to specify others. Some examples of missing parameters are the number of iterations used for reference selection and the chunk sizes (used for outdegrees encoding). Speaking about these two, in my implementation of Zuckerli I have not added the option of doing multiple iterations of reference selection, thus fixing that parameter to _1_. Moreover, since I didn't want to radically change the existing code structure, I didn't implement the outdegrees chunk part, thus forcing the chunk size _C_ (as they refer to it in the paper) to _∞_.
Finally, if you want to run both the C++ version and the Rust one, ensure that you are setting a window size _W_ of _32_, and that you are using Zuckerli encoding's parameters _k, i,_ and _j_ with values _4, 1, 0_, respectively.

**Comment by vigna** [comment]
> Hi @vigna and @colobrodo, sorry for the late reply.
> Unfortunately, there is no reproducibility script, as the main aim of the project was not to prove nor disprove the results published in the authors' papers, but instead to do some experiments on them and mainly to rewrite them in Rust, testing the language capabilities.

Well, if you run any kind of experiment, and you want them to be useful for others, replication is necessary, and a script is always a good idea. In fact it could be also useful for you, as you might forget details of your experiments. 

Anyway, thanks for the parameters, we'll try to replicate and get back to you if we need more help.

**Comment by colobrodo** [comment]
Hi, 
I have tried to reproduce your results, but my results are different. I have used a window size of 32 and a max reference count of 3, but I have kept the parameters `i`, `j` and `k` for hybrid integer encoding to the defaults of 4, 2 and 1, as I see these are hardcoded constants both here and in c++.
The results I get are quite different from the authors' version: the graphs are in some cases larger than the original webgraph version. I write a script to download a graph (`cnr-2000`, the smallest), generate the offset and convert it using the resulting binaries compiling your work in release mode. 

```bash
#!/bin/bash

if [ ! -d cnr-2000 ]; then
        mkdir cnr-2000
fi

if [ ! -d cnr-2000-zuckerli ]; then
        mkdir cnr-2000-zuckerli
fi

cd cnr-2000

for ext in .properties .graph .md5sums; do
        wget -c http://data.law.di.unimi.it/webdata/cnr-2000/cnr-2000$ext
done

cd ..
WebGraph/target/release/generate_offsets cnr-2000/cnr-2000
WebGraph/target/release/compress_zuckerli -w 32 cnr-2000/cnr-2000 cnr-2000-zuckerli/cnr-2000

```

I have converted the same graph with the cpp version and the same parameter and obtained a bits per edge close to the one declared by the author in the paper, while the one generated by the script is bigger than the one downloaded (1.4M vs 1.2M).
Why are the results so different? is there something wrong in my reproduction of the results?

Thanks
